gprofiler/metadata/application_identifiers.py

```diff
@@ -0,0 +1,256 @@
+#
+# Copyright (c) Granulate. All rights reserved.
+# Licensed under the AGPL3 License. See LICENSE.md in the project root for license information.
+#
+import configparser
+import os.path
+import re
+from abc import ABCMeta, abstractmethod
+from typing import List, Optional
+
+from granulate_utils.linux.ns import resolve_host_path
+from psutil import NoSuchProcess, Process
+
+from gprofiler.log import get_logger_adapter
+
+_logger = get_logger_adapter(__name__)
+
+
+_PYTHON_BIN_RE = re.compile(r"^python([23](\.\d{1,2})?)?$")
+
+
+# Python does string interning so just initializing str() as a sentinel is not enough.
+class StrSentinel(str):
+    pass
+
+
+_NON_AVAILABLE_ARG = StrSentinel()
+
+
+def _is_python_m_proc(process: Process) -> bool:
+    """
+    Checks whether the process ran as "python -m ..." pattern.
+    """
+    args = process.cmdline()
+
+    # Just to be on the safe side, module name must be run after the "-m" argument, python enforces it.
+    if len(args) < 3:
+        return False
+
+    return _is_python_bin(args[0]) and args[1] == "-m"
+
+
+def _is_python_bin(bin_name: str) -> bool:
+    return _PYTHON_BIN_RE.match(os.path.basename(bin_name)) is not None
+
+
+def _get_cli_arg_by_name(args: List[str], arg_name: str, check_for_equals_arg: bool = False) -> str:
+    if arg_name in args:
+        return args[args.index(arg_name) + 1]
+
+    if check_for_equals_arg:
+        for arg in args:
+            arg_key, _, arg_val = arg.rpartition("=")
```

Comment:
```suggestion
            arg_key, _, arg_val = arg.partition("=")
```

So for `--blabla=a=c=b` the value is `a=c=b`